transformations/transformations/data_offload.py

```diff
@@ -370,7 +369,7 @@ def process_driver(self, routine, successors):
             'enter_data_create': 'enter data create',
         }
         for key, directive in key_directive_map.items():
-            variables = set.union(*[s.trafo_data.get(self._key, {}).get(key) for s in successors], set())
+            variables = set.union(*[s.trafo_data.get(self._key, {}).get(key, {}) for s in successors], set())
```

Comment:
This change avoids an iteration error over `None`s, since in the original version, the `get(key)` can return a `None`